### DISCUSSION TITLE
Controlling video playback

### DISCUSSION THREAD

**Discussion by ricardogsilva** [cables-gl/cables_docs#872] [Answers & Questions] [answered]
I'd like to be able to control the current frame of a video texture in order to build some sort of video scratching patch. Is this possible?

Being fairly new to cables, my naive approach was to checkout the `VideoTexture` op and hope that there would be some parameter to specify the current frame, or perhaps just being able to specify the playback speed and direction.

I could only find the `speed` parameter, which only works for positive numbers, so it seems I cannot make the video playback in reverse

![image](https://github.com/user-attachments/assets/b5979361-f63e-413e-8873-15522133d86d)

**Comment by 60-hz** [comment, marked as answer]
Set Time is what you are looking for. It is relative to length of video.

But...

To have a smooth playback, you need intraframe-only encoded movie file. In this type of encoding, each frame is encoded as a standalone full frame (a keyframe), without referencing other frames for compression. This is in contrast to interframe compression, where some frames (non-keyframes) only store changes relative to other frames (e.g., P-frames and B-frames). 

Doing this on the web seems to be a little bit tricking due to some player limitation.
I want to to this also and really think we are close to it... 

[Here is a patch and issue about it](https://github.com/cables-gl/cables_docs/issues/842). You need to wait untill the player load all frames to be able to do it, but it work at the end for me. 

In the standalone version of cables, there is currently a bug with ffmpeg when opening some .mp4 movies, hoping it will be fixed soon...

**Reply by ricardogsilva** [reply]
Thanks, very informative - I also checked out your patch, which seems to work nicely!
